emscripten.py

```diff
@@ -793,8 +793,9 @@ def create_receiving(exports):
         mangled = asmjs_mangle(s)
         dynCallAssignment = ('dynCalls["' + s.replace('dynCall_', '') + '"] = ') if generate_dyncall_assignment and mangled.startswith('dynCall_') else ''
         export_assignment = ''
-        if settings.MODULARIZE and settings.EXPORT_ALL:
-          export_assignment = f'Module["{mangled}"] = '
+        if settings.MODULARIZE:
+          if settings.EXPORT_ALL or (settings.EXPORT_KEEPALIVE and mangled in settings.EXPORTED_FUNCTIONS):
```

Comment:
Extracting `should_export` should also allow this to be a single `if` statement again.

Reply:
Don't mind doing that but it means that the conditional is then evaluated even in non-modularize case, which isn't needed

Reply:
Maybe make `should_export()` a function then if you prefer?